env0/resource_environment.go

```diff
@@ -286,6 +286,23 @@ func setEnvironmentConfigurationSchema(d *schema.ResourceData, configurationVari
 	}
 }
 
+// Validate that the template is assigned to the "project_id".
+func validateTemplateProjectAssignment(d *schema.ResourceData, apiClient client.ApiClientInterface) error {
+	projectId := d.Get("project_id").(string)
+	templateId := d.Get("template_id").(string)
+
+	template, err := apiClient.Template(templateId)
```

Comment:
so in case of the race condition described [here](https://github.com/env0/terraform-provider-env0/issues/478) if we get here and we didn't assign the template yet it would throw an exception here?

Reply:
@ItamarMalka 
I don't think so. The template must exist irrespective of the project assignment.